pandas/core/strings/base.py

```diff
@@ -0,0 +1,229 @@
+import abc
+from typing import Pattern, Union
+
+import numpy as np
+
+from pandas._typing import Scalar
+
+from pandas.core.arrays.base import ExtensionArray
+
+
+class BaseStringArrayMethods(abc.ABC):
+    """
+    Base class for array _str accessor.
+
+    This is where ExtensionArrays can override the implementation of
```

Comment:
```suggestion
    This is where our ExtensionArrays can override the implementation of
```

(that's maybe clearer that it's internal (for now) ..)